components/Cards/tests/snapshot/MDCCardCustomTraitCollectionSnapshotTests.m

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialSnapshot.h"
+
+#import "MaterialCards+Theming.h"
+#import "MaterialCards.h"
+
+@interface MDCCardWithCustomTraitCollection : MDCCard
+@property(nonatomic, strong) UITraitCollection *traitCollectionOverride;
+@end
+
+@implementation MDCCardWithCustomTraitCollection
+- (UITraitCollection *)traitCollection {
+  return self.traitCollectionOverride ?: [super traitCollection];
+}
+@end
+
+@interface MDCCardCustomTraitCollectionSnapshotTests : MDCSnapshotTestCase
+
+@property(nonatomic, strong) MDCContainerScheme *containerScheme;
+@property(nonatomic, strong) MDCCardWithCustomTraitCollection *card;
+
+@end
+
+@implementation MDCCardCustomTraitCollectionSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  //    self.recordMode = YES;
+
+  self.card = [[MDCCardWithCustomTraitCollection alloc] initWithFrame:CGRectMake(0, 0, 80, 80)];
+  self.containerScheme = [[MDCContainerScheme alloc] init];
+  self.containerScheme.colorScheme =
+      [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
+}
+
+- (void)tearDown {
+  self.card = nil;
+  self.containerScheme = nil;
+
+  [super tearDown];
+}
+
+#pragma mark - Helpers
+
+- (void)generateSnapshotAndVerifyForView:(UIView *)view {
+  UIView *snapshotView = [view mdc_addToBackgroundView];
+  [self snapshotVerifyView:snapshotView];
+}
+
+- (void)testShadowColorRespondsToDynamicColor {
+#if defined(__IPHONE_13_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_0)
+  if (@available(iOS 13.0, *)) {
+    // Given
+    UIColor *darkModeColor = UIColor.whiteColor;
+    UIColor *dynamicColor =
+        [UIColor colorWithDynamicProvider:^(UITraitCollection *traitCollection) {
+          if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleLight) {
+            return UIColor.blackColor;
+          } else {
+            return darkModeColor;
```

Comment:
Please make the shadow color something very obviously different from the normal/default value. I would suggest something like `UIColor.magentaColor`.  I would also recommend increasing the elevation of the card so the shadow is easier to see in the image.